### PR TITLE
Add a flag to NoSqlProvider to skip ordering results. This can make th…

### PR DESCRIPTION
**Pull request by berickson1**
…e sqlite provider faster on large datasets

Keep backwards compatibility with the existing boolean flag

### CODE REVIEW

File: src/IndexedDbProvider.ts [hunk 1]
```diff
@@ -720,22 +720,24 @@ class IndexedDbIndex extends FullTextSearchHelpers.DbIndexFTSFromRangeQueries {
         }
     }
 
-    getAll(reverse?: boolean, limit?: number, offset?: number): SyncTasks.Promise<ItemType[]> {
+    getAll(reverseOrSortOrder?: boolean | NoSqlProvider.QuerySortOrder, limit?: number, offset?: number): SyncTasks.Promise<ItemType[]> {
         // ************************* Don't change this null to undefined, IE chokes on it... *****************************
         // ************************* Don't change this null to undefined, IE chokes on it... *****************************
         // ************************* Don't change this null to undefined, IE chokes on it... *****************************
+        const reverse = reverseOrSortOrder === false || reverseOrSortOrder === NoSqlProvider.QuerySortOrder.Reverse;
```
Comment (ms-markda) [inline src/IndexedDbProvider.ts]: Should this `false` be `true` instead? [For all 3 usages in this file.] This is different from `InMemoryProvider.ts`.

Comment (ms-markda) [inline src/IndexedDbProvider.ts]: Nit: Move `const` above comment. The farther the comment is from the `null!!!` then the less likely the reader will notice and read the comment.

File: src/tests/NoSqlProviderTests.ts [hunk 2]
```diff
@@ -284,6 +284,14 @@ describe('NoSqlProvider', function () {
                                 [4].forEach(v => { assert(_.find(ret, r => r.val === 'val' + v)); });
                             });
 
+                        let t3b3 = prov.getRange('test', indexName, formIndex(2), formIndex(4), false, false,
```
Comment (ms-markda) [inline src/tests/NoSqlProviderTests.ts]: Should you also add tests for `Forward` and `None`?
